service/templates/projects.html

```diff
@@ -633,7 +633,49 @@ <h3>Periodic Builds</h3>
             </div>
           </div>
         </div>
+        <!-- Commands for Virtual Workstations -->
+        <div class="variables">
+          <div class="form-group">
+            <div class="col-header col-lg-6 form-control-static">
+              <h3>Virtual Workstations Commands</h3>
+              <div class="muted small">Specify a command to be run on a virtual workstation for this project, and optionally a subdirectory to run it in. If using git clone, the repo and branch will be taken from this project.</div>
+            </div>
+          </div>
+
+          <div id="workstation-header" class="form-group">
+            <div class="col-lg-4"> <label class="control-label"> Command </label> </div>
+            <div class="col-lg-2"> <label class="control-label"> Directory </label> </div>
+            <div class="col-lg-2"></div>
+          </div>
 
+          <div id="workstation-list" class="form-group" ng-repeat="obj in settingsFormData.workstation_config track by $index">
+            <div class="col-lg-4">
+              <textarea class="form-control" ng-model="obj.command" placeholder="command"></textarea>
+            </div>
+            <div class="col-lg-2">
+              <input class="form-control" ng-model="obj.directory" type="text" placeholder="directory">
+            </div>
+            <div class="col-lg-2">
+              <button class="btn btn-default btn-danger" type="button" ng-click="removeWorkstationCommand($index)">
+                <i class="fa fa-trash"></i>
+              </button>
+            </div>
+          </div>
+
+          <div class="form-group">
+            <div class="col-lg-4">
+              <textarea class="form-control" ng-model="workstation_command.command" placeholder="command"></textarea>
+            </div>
+            <div class="col-lg-2">
+              <input ng-model="workstation_command.directory" class="form-control" type="text" placeholder="directory">
+            </div>
+            <div class="col-lg-2">
+              <button class="plus-button btn btn-primary " ng-disabled="!workstation_command.command" type="button" ng-click="addWorkstationCommand()">
```

Comment:
It's pretty much the same, but should this actually use the validation function to check?